File: ConcordApi/Models/Channel.cs
namespace ConcordApi.Models;

public class Channel
{
    public int Id { get; set; }
    public string Name { get; set; }
    
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public ICollection<Message> Messages { get; set; } = new List<Message>();
}

File: ConcordApp/src/lib/models/channel.ts
import {Message} from "@/lib/models/message";

export class Channel {
    id: number;
    name: string;

    createdAt: Date;
    updatedAt: Date;

    messages: Message[];

    constructor(partial?: Partial<Channel>) {
        Object.assign(this, partial);
    }
}
